code/datums/components/thermite.dm

```diff
@@ -0,0 +1,72 @@
+/datum/component/thermite
+	var/static/list/blacklist
+	var/static/list/resistlist
+	var/amount
+
+/datum/component/thermite/Initialize(_amount)
+	. = ..()
+	blacklist = typecacheof(/turf/closed/wall/mineral/diamond)
```

Comment:
this can be made in the class declaration(otherwise it will happen in every initialize)

Reply:
For some reason at the time I was thinking you can't use typecacheof in declaration